controllers/controllers/cluster_controller_v2.go

```diff
@@ -0,0 +1,295 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/controllers/remote"
+	capicontrolplane "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/util/patch"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+)
+
+// ClusterReconcilerV2 reconciles a Cluster object
+type ClusterReconcilerV2 struct {
+	Client  eksaClient
+	Log     logr.Logger
+	Scheme  *runtime.Scheme
+	Tracker *remote.ClusterCacheTracker
+}
+
+func NewClusterReconcilerV2(client client.Client, tracker *remote.ClusterCacheTracker, log logr.Logger, scheme *runtime.Scheme) *ClusterReconcilerV2 {
+	return &ClusterReconcilerV2{
+		Client:  eksaClient{Client: client},
+		Log:     log,
+		Scheme:  scheme,
+		Tracker: tracker, // TODO: Need to figure what to do to avoid: {"reconciler group": "anywhere.eks.amazonaws.com", "reconciler kind": "Cluster", "name": "br-w3-cluster", "namespace": "eksa-system", "error": "the cache is not started, can not read objects"}
+	}
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ClusterReconcilerV2) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&anywherev1.Cluster{}). // TODO: add filtering to ignore self managed clusters
+		Watches(&source.Kind{Type: &capi.Cluster{}},
```

Comment:
what's the reason to watch capi resources instead of eksa resources? and do we also need to watch provider specific resources?

Reply:
Good question
It's not instead but in addition. We are not watching other eks-a resources yet bc the cluster doesn't react to "spec" changes (eg. it doesn't support upgrades). The final version will need to watch at least datacenter and machine config objects

As for why we need to watch capi resources: that's how we "wait" for certain conditions in the capi objects before moving to the next step of our reconciliation. We watch those resources and enqueue a `anywhere.Cluster` request for the proper cluster. That way we can react to changes in those resources without blocking the reconcile thread.